pennylane/ops/qubit.py

```diff
@@ -1650,16 +1674,104 @@ def __init__(self, *params, control_wires=None, wires=None, do_queue=True):
 
         wires = control_wires + wires
 
-        # Given that the controlled wires are listed before the target wires, we need to create a
-        # block-diagonal matrix of shape ((I, 0), (0, U)) where U acts on the target wires and I
-        # acts on the control wires.
-        padding = 2 ** len(wires) - len(U)
-        CU = block_diag(np.eye(padding), U)
+        # If unspecified, we control on the all-ones string
+        if not control_values:
+            control_values = 2 ** len(control_wires) - 1
+
+        control_int = self._parse_control_values(control_wires, control_values)
+
+        # A multi-controlled operation is a block-diagonal matrix partitioned into
+        # blocks where the operation being applied sits in the block positioned at
+        # the integer value of the control string. For example, controlling a
+        # unitary U with 2 qubits will produce matrices with block structure
+        # (U, I, I, I) if the control is on bits '00', (I, U, I, I) if on bits '01',
+        # etc. The positioning of the block is controlled by padding the block diagonal
+        # to the left and right with the correct amount of identity blocks.
+
+        padding_left = control_int * len(U)
+        padding_right = 2 ** len(wires) - len(U) - padding_left
+
+        CU = block_diag(np.eye(padding_left), U, np.eye(padding_right))
 
         params = list(params)
         params[0] = CU
+
         super().__init__(*params, wires=wires, do_queue=do_queue)
 
+    @staticmethod
+    def _parse_control_values(control_wires, control_values):
+
+        control_int = control_values
```

Comment:
Perhaps this statement can be put inside the `if isinstance(control_values, int)`?